docs/storage-abstraction.md

```diff
@@ -1,35 +1,57 @@
-# Storage abstraction
+# Storage Abstraction
```

Comment:
Since many parts of this doc explain the Storage API, should we rename the title to something like the following?:

```suggestion
# Storage Abstraction and API Guide
```

Reply:
Agreed!